code/game/objects/items/devices/portable_recharger.dm

```diff
@@ -0,0 +1,330 @@
+	return ..()
+
+/obj/item/portable_recharger/Exited(atom/movable/gone, direction)
+	if(gone == charging)
+		if(!QDELING(charging))
+			charging.update_appearance()
+		charging = null
+		using_power = FALSE
+		soundloop.stop()
+		update_appearance()
+	return ..()
+
+/obj/item/portable_recharger/attackby(obj/item/attacking_item, mob/user, params)
+	if(hit)
+		return
+
+	if(istype(attacking_item, /obj/item/stock_parts/cell))
+		if(cell)
+			to_chat(user, span_warning("[src] already has a cell!"))
+		else
+			if(!user.transferItemToLoc(attacking_item, src))
+				return TRUE
+			cell = attacking_item
+			to_chat(user, span_notice("You install a cell in [src]."))
+
+	if(!cell)
+		balloon_alert(user, "no cell!")
+		return ..()
+
+	if(!is_type_in_typecache(attacking_item, allowed_devices))
+		return ..()
+
+	if(charging)
+		return TRUE
+
+	if(!cell.charge)
+		to_chat(user, span_notice("[src] blinks red as you try to insert [attacking_item]."))
+		return TRUE
+
+	if(istype(attacking_item, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = attacking_item
+		if(!energy_gun.can_charge)
+			to_chat(user, span_notice("Your gun has no external power connector."))
+			return TRUE
+	user.transferItemToLoc(attacking_item, src)
+	return TRUE
+
+/obj/item/portable_recharger/screwdriver_act(mob/living/user, obj/item/tool)
+	if(!cell || charging || hit)
+		return FALSE
+
+	if(do_after(user, 3 SECONDS, target = src))
+		cell.update_appearance()
+		cell.forceMove(get_turf(src))
+		balloon_alert(user, "removed [cell]")
+		cell = null
+		tool.play_tool_sound(src, 50)
+		return TRUE
+	return FALSE
+
+/obj/item/portable_recharger/attack_hand(mob/user, list/modifiers)
+	if(loc == user)
+		if(slot_flags & ITEM_SLOT_BACK)
+			if(user.get_item_by_slot(ITEM_SLOT_BACK) == src)
+				take_charging_out(user)
+			else
+				to_chat(user, span_warning("Put the [src] on your back first!"))
+
+		else if(slot_flags & ITEM_SLOT_BELT)
+			if(user.get_item_by_slot(ITEM_SLOT_BELT) == src)
+				take_charging_out(user)
+			else
+				to_chat(user, span_warning("Strap the [src] on first!"))
+		return
+
+	add_fingerprint(user)
+	return ..()
+
+/obj/item/portable_recharger/MouseDrop(obj/over_object)
+	. = ..()
+	if(ismob(loc))
+		var/mob/M = loc
+		if(!M.incapacitated() && istype(over_object, /atom/movable/screen/inventory/hand))
+			var/atom/movable/screen/inventory/hand/H = over_object
+			M.putItemFromInventoryInHandIfPossible(src, H.held_index)
+
+///Takes charging item out if there is one
+/obj/item/portable_recharger/proc/take_charging_out(mob/user)
+	if(isnull(charging) || user.put_in_hands(charging))
+		return
+	charging.forceMove(drop_location())
+
+/obj/item/portable_recharger/attack_tk(mob/user)
+	if(isnull(charging))
+		return
+	charging.forceMove(drop_location())
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+///Blows up the cell in it
+/obj/item/portable_recharger/proc/detonate()
+	if(!cell)
+		return
+
+	cell.explode()
+
+	if(charging)
+		charging.forceMove(drop_location())
```

Comment:
```suggestion
	charging?.forceMove(drop_location())
```
But also: You should implement `deconstruct()` on this item, where it dumps anything inside it on the ground, and then instead of calling `qdel(src)` call `deconstruct()`